File: iteracion2/modules/dispositivo/src/DispositivoModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class DispositivoModule extends NativeModule<{}> {
  getModelo(): string;
  getFabricante(): string;
  getVersionAndroid(): string;
  saludar(nombre: string): string;
}

export default requireNativeModule<DispositivoModule>('Dispositivo');

File: iteracion2/modules/dispositivo/android/src/main/java/expo/modules/dispositivo/DispositivoModule.kt
package expo.modules.dispositivo

import android.os.Build
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class DispositivoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("Dispositivo")

    Function("getModelo") {
      Build.MODEL
    }

    Function("getFabricante") {
      Build.MANUFACTURER
    }

    Function("getVersionAndroid") {
      "Android ${Build.VERSION.RELEASE} (API ${Build.VERSION.SDK_INT})"
    }

    Function("saludar") { nombre: String ->
      "¡Hola, $nombre! Este mensaje viene de código Kotlin nativo en Android."
    }
  }
}
